### ISSUE TITLE
Distance - App redirects to expense report when returning online after editing distance offline

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33622]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.17-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

**Issue found when executing PR** https://github.com/Expensify/App/pull/33266
## Action Performed:
Precondition: There are three addresses saved in Recent destinations
1. Create a distance request with address A and B as Start and Finish points respectively
2. Navigate to the details page of the distance request
3. Go offline
4. Click Distance > Add stop > Add address C > Save it
5. Click Distance > Address C (Finish point) > Edit it so that Finish point becomes address A too
6. Save the edit
7. Go online

## Expected Result:
App will stay in distance request details page

## Actual Result:
App redirects to expense report automatically

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>  

https://github.com/Expensify/App/assets/78819774/82026aae-d697-4207-9e6c-88d078dcaa48

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by tgolen** [comment]
I'll start investigating. It's possible it was caused by https://github.com/Expensify/App/pull/33266 but I want to verify that.

**Comment by mountiny** [comment]
@neil-marcellini @allroundexperts @dukenv0307 This is linked to PR you have been involved in, it does seem though that this is already in production if I am not mistaken and given its edge case I would vote this is not a blocker. @tgolen let me know what you think?

**Comment by tgolen** [comment]
I reverted the changes from https://github.com/Expensify/App/pull/33266 locally, and I had trouble reproducing the issue because there is a bug that the PR fixes where you can't save the request in offline mode (clicking the button does nothing). So, I'm not so sure that PR caused the redirect issue.

I do agree with @mountiny that the redirect issue reported here is not bad enough to be a deploy blocker. Everything works, it's just you end up in a slightly different place and it's easy to go back to the details.

I'm removing the deploy blocker label and I think we should consider opening this bug up to contributors to find the cause of it.

**Comment by paultsimura** [comment]
I believe this is the same issue as with the scan receipts flow: the `transactionThreadReportID` in the API response (in our case – CreateDistanceRequest) comes different from the one created optimistically. So when we come back online, and the CreateDistanceRequest API Response arrives, the opened expense details report no longer exists.

It should be fixed with https://github.com/Expensify/App/issues/31411 (we are holding https://github.com/Expensify/App/issues/32552 for the same reason).

**Comment by bfitzexpensify** [comment]
Thanks @paultsimura. Subscribed to https://github.com/Expensify/App/issues/31411, will re-test this once that's done.

**Comment by tgolen** [comment]
This is still on hold since https://github.com/Expensify/App/issues/31411 is not done.

**Comment by tgolen** [comment]
Still on hold

**Comment by tgolen** [comment]
Still on hold

**Comment by bfitzexpensify** [comment]
Still on hold for https://github.com/Expensify/App/issues/31411 - looks like the [PR](https://github.com/Expensify/App/pull/35449) is in the final review stretch

**Comment by bfitzexpensify** [comment]
Same status - the PR we're waiting on has had all comments resolved in the last 24 hours, so we're close to being able to retest here.

**Comment by bfitzexpensify** [comment]
Still waiting on https://github.com/Expensify/App/issues/31411

**Comment by bfitzexpensify** [comment]
Asked in [Slack](https://expensify.slack.com/archives/C9YU7BX5M/p1709622133424929) for this to be retested

**Comment by bfitzexpensify** [comment]
Thanks for the heads-up @paultsimura, I'll get QA to pause on the retesting for the moment then.

**Comment by bfitzexpensify** [comment]
#33114 is live - getting QA to retest this.

**Comment by kavimuru** [comment]
@bfitzexpensify Issue is not reproducible by the tester. 

https://github.com/Expensify/App/assets/43996225/ed4449d9-a7fc-413c-846c-2fcc5340c4fa

**Comment by bfitzexpensify** [comment]
Great! Let's close this one out then.
